controllers/helpers.go

```diff
@@ -777,5 +787,171 @@ func getPrismCentralClientForCluster(ctx context.Context, cluster *infrav1.Nutan
 	}
 
 	conditions.MarkTrue(cluster, infrav1.PrismCentralClientCondition)
-	return v3Client, nil
+	return client, nil
+}
+
+func getPrismCentralV4ClientForCluster(ctx context.Context, cluster *infrav1.NutanixCluster, secretInformer v1.SecretInformer, mapInformer v1.ConfigMapInformer) (*prismclientv4.Client, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	clientHelper := nutanixclient.NewHelper(secretInformer, mapInformer)
+	managementEndpoint, err := clientHelper.BuildManagementEndpoint(ctx, cluster)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("error occurred while getting management endpoint for cluster %q", cluster.GetNamespacedName()))
+		conditions.MarkFalse(cluster, infrav1.PrismCentralClientCondition, infrav1.PrismCentralClientInitializationFailed, capiv1.ConditionSeverityError, err.Error())
+		return nil, err
+	}
+
+	client, err := nutanixclient.NutanixClientCacheV4.GetOrCreate(&nutanixclient.CacheParams{
+		NutanixCluster:          cluster,
+		PrismManagementEndpoint: managementEndpoint,
+	})
+	if err != nil {
+		log.Error(err, "error occurred while getting nutanix prism client from cache")
+		conditions.MarkFalse(cluster, infrav1.PrismCentralClientCondition, infrav1.PrismCentralClientInitializationFailed, capiv1.ConditionSeverityError, err.Error())
+		return nil, fmt.Errorf("nutanix prism client error: %w", err)
+	}
+
+	conditions.MarkTrue(cluster, infrav1.PrismCentralClientCondition)
+	return client, nil
+}
+
+// isPrismCentralV4Compatible checks if the Prism Central is v4 API compatible
+func isPrismCentralV4Compatible(ctx context.Context, v3Client *prismclientv3.Client) (bool, error) {
+	pcVersion, err := getPrismCentralVersion(ctx, v3Client)
+	if err != nil {
+		return false, fmt.Errorf("failed to get Prism Central version: %w", err)
+	}
+
+	// Check if the version is v4 compatible
+	// PC versions look like pc.2024.1.0.1
+	// We can check if the version is greater than or equal to 2024
```

Comment:
Issue: this handles the production PC versions but not the dev and internal versions e.g. `master`, `fraser-2024.1-stable-pc-0`